src/wallet/db.cpp

```diff
@@ -550,17 +550,17 @@ bool BerkeleyBatch::Rewrite(BerkeleyDatabase& database, const char* pszSkip)
                         fSuccess = false;
                     }
 
-                    Dbc* pcursor = db.GetCursor();
-                    if (pcursor)
+                    if (db.CreateCursor())
```

Comment:
In commit "walletdb: Handle cursor internally" (2711caa2a2d4f04b848c25b614b353a1a08351b0)

Behavior isn't changing in this commit, but error handling here is broken, right? This should be returning failure if creating a cursor doesn't succeed, not just pretending it succeeded and not reading anything.

Reply:
Yes, I think error handling is broken here. I think the intent was to ignore an error and just keep trying until it succeeds. But there's nothing to actually have the outer while loop keep going.